### ISSUE TITLE
Running commands with Typescript is slow

### ISSUE THREAD

**Issue by mploux** [tj/commander.js#1633]
Hello all ! :wave:

First of all, I love this package, so thanks a lot :muscle:

I was wondering if I was the only one getting very slow execution times using typescript. 
I am asking here to get advise and/or help on making my cli faster.

In the doc, it is written that for typescript project we should execute the cli through `node` as follow:
```
$ node -r ts-node/register cli.ts
```
Whereas the most common way of executing typescript is with `ts-node` directly.
We need to execute through `node` in order to correctly load each sub-command. 
In fact, for me I need to do so, because sub-commands indeed wont execute.

The problem is that running typescript through `node` is about 10x slower than with `node` for me.

I did a benchmark:

Running with `ts-node` takes around 0.6secs :
![Screenshot_22](https://user-images.githubusercontent.com/17877331/139542409-caa93da9-435c-49c7-b44a-e89f25d5504e.jpg)
 
Running through `node` takes around 5secs :
![Screenshot_23](https://user-images.githubusercontent.com/17877331/139542413-b45f125b-f832-496d-aa6c-a0c01810d054.jpg)

In conclusion, for me, the cli is really slow to use, and close to unusable in some cases. It slows everything having to wait 5 seconds for each command.

Is there some sort of way to use `ts-node` **and** have sub-commands working ?
If not, do you know of any optimizations I could use ?

My configuration: Win10 - WSL1 - Debian v11 - Using fish - node v16.13.0 - ts-node v10.4.0 - TS v4.4.4

I feel like it's a WSL problem, as another dev got the same issue on WSL: #955 

Thanks a lot !

**Comment by shadowspawn** [comment]
I have some suggestions, but first a couple of clarifications.

1) using `node -r ts-node/register` is only suggested if you want to use `ts-node` on the `.ts` file (rather than compile the source first), AND you have implemented your subcommands as stand-alone executable `.ts` files (rather than using action handlers). 

2) Your timing test is not directly comparing `ts-node` and `node -r ts-node/register`. Both of your examples have something different.

a) The first test is interesting on its own. You are running `ts-node -T` rather than just `ts-node`.

From the command-line help:

```
    -T, --transpile-only            Use TypeScript's faster `transpileModule` or a third-party transpiler
```

b) In the second test you are running `ts-node` instead of `node` with the require option. That combination isn't one we are interested in!

i.e. you have called `ts-node -r ts-node/register` rather than `node -r ts-node/register`.

For my laptop (macOS), I see timings like:

```
./node_modules/.bin/ts-node index.ts
2.02s user 0.10s system 180% cpu 1.176 total

node --require ts-node/register index.ts
2.02s user 0.10s system 183% cpu 1.159 total

./node_modules/.bin/ts-node --transpile-only index.ts
0.31s user 0.04s system 100% cpu 0.342 total

node vanilla.js
0.05s user 0.01s system 100% cpu 0.067 total
```

So for me calling `node -r ts-node/register` is similar speed to calling `ts-node`. What is faster is using `transpile-only` (And what is much faster again is using node with JavaScript rather than ts-node with TypeScript!)

**Comment by shadowspawn** [comment]
Suggestion 1

The `ts-node` README has an example call of `node -r ts-node/register/transpile-only` to get the `transpile-only` behaviour. For running TypeScript faster that might be an easy win!

- https://github.com/TypeStrong/ts-node#programmatic
- https://github.com/TypeStrong/ts-node#third-party-transpilers

**Comment by mploux** [comment]
I was just about to look how to use `transpile-only` using `node` 
Thanks I lot it was exactly what I was looking for, in fact an easy win :muscle:

Keep up the amazing work and thanks for the fast response !

**Comment by shadowspawn** [comment]
Suggestion 2

Compile the TypeScript and run the JavaScript!

Just pointing out explicitly for future readers that a work-around for "running commands with TypeScript is slow" is, don't run commands with TypeScript. 😄

**Comment by mploux** [comment]
Yes I 100% agree with you ! 
In my case it's not practical because I need to use de CLI in runtime, but for many other developers it should be the solution !
Thanks again 💪
